src/Halogen/HTML/Events/Types.purs

```diff
@@ -33,6 +36,23 @@ type MouseEvent =
   , which :: Number
   )
 
+-- | Identifies the additional fields which are available on drag events.
+type DragEvent =
+  ( button :: Number
+  , detail :: Number
+  , relatedTarget :: HTMLElement
+  , clientX :: Number
+  , clientY :: Number
+  , screenX	:: Number
+  , screenY	:: Number
+  , ctrlKey	:: Boolean
+  , shiftKey :: Boolean
+  , altKey :: Boolean
+  , metaKey	:: Boolean
+  , which :: Number
+  , dataTransfer :: Nullable DataTransfer
```

Comment:
Yes, the question is where to put `toMaybe`. We could just define the handlers like:
```haskell
onDrop f =
  handler
    (eventName "drop")
    \ev ->
      f { button: ev.button
        , detail: ev.detail
        ...
        , dataTransfer: Nullable.toMaybe ev.dataTransfer
        } 
```